File: leetcode/392.cs
public class Solution {
    public bool IsSubsequence(string s, string t) {
        if(s.Length == 0) return true;
        int curPosS = 0;
        for(int i = 0; i < t.Length; i++){
            if(s[curPosS] == t[i]){
                curPosS++;
                if(curPosS == s.Length){
                    return true;
                }
            }
            
        }
        
        return false;
    }
}

File: leetcode/392.ts
function isSubsequence(s: string, t: string): boolean {
    let sPointer = 0;
    let tPointer = 0;
    while(tPointer < t.length && sPointer < s.length){
        if(s.charAt(sPointer) === t.charAt(tPointer)){
            sPointer++;
            tPointer++;
        }
        else{
            tPointer++;
        }
    }
    return sPointer === s.length;
};
